### DISCUSSION TITLE
Why do we need this plugin?

### DISCUSSION THREAD

**Discussion by albgen** [nonzzz/vite-plugin-compression#43] [General]
as per title, why we need it?
Usually is the web server job to compress, cache and deliver this content. Why a plugin to produce a compressed file?

**Comment by nonzzz** [comment]
Since it needs to be processed on the server side, it will increase the overhead of the server side.

**Comment by albgen** [comment]
The prcess of zip is done only once on the webserver more specifically only on the first request of a resource. After that, it can be cached and no need to zip again the same resource because the zipped version will be served. So basically just to avoid this first overhead on compression of the webserver?

**Reply by nonzzz** [reply]
yeap

**Comment by thednp** [comment]
Does it also compress HTML? I'm getting "Enable text compression - Potential savings of 13 KiB" issue in Lighthouse so I wonder if this plugin helps with that (I'm developing currently on SolidStart / Vinxi with Nitro and Vite).

😄 Happy New Year! 😄

**Reply by nonzzz** [reply]
Happy new year too,This plugin can using zlib to compress anything unless you set special exclude or include 😃
